udata/features/territories/commands.py

```diff
@@ -50,70 +53,82 @@ def migrate_zones_ids():
-                        counter_counties += 1
-                        new_zones.append(
-                            GeoZone
-                            .objects(code=zone_id, level='fr:departement')
-                            .valid_at(date.today())
-                            .first())
-                    elif kind == 'region':
-                        counter_regions += 1
-                        # Only link to pre-2016 regions which kept the same id.
-                        new_zones.append(
-                            GeoZone
-                            .objects(code=zone_id, level='fr:region')
-                            .first())
-                    else:
-                        new_zones.append(zone)
-                elif zone.id.startswith('country-subset/fr'):
-                    counter_zones += 1
-                    subset, country, kind = zone.id.split('/')
-                    if kind == 'dom':
-                        counter_drom += 1
-                        new_zones.append(drom_zone)
-                    elif kind == 'domtom':
-                        counter_dromcom += 1
-                        new_zones.append(dromcom_zone)
-                elif zone.id.startswith('country/fr'):
-                    counter_zones += 1
-                    counter_france += 1
-                    new_zones.append(france_zone)
+    # Iter over datasets with zones
+    for dataset in Dataset.objects(spatial__zones__gt=[]):
+        counter['datasets'] += 1
+        new_zones = []
+        for zone in dataset.spatial.zones:
+            if zone.id.startswith('fr/'):
+                counter['zones'] += 1
+                country, kind, zone_id = zone.id.split('/')
+                zone_id = zone_id.upper()  # Corsica 2a/b case.
+                if kind == 'town':
+                    counter['towns'] += 1
+                    new_zones.append(
+                        GeoZone
+                        .objects(code=zone_id, level='fr:commune')
+                        .valid_at(date.today())
+                        .first())
+                elif kind == 'county':
+                    counter['counties'] += 1
+                    new_zones.append(
+                        GeoZone
+                        .objects(code=zone_id, level='fr:departement')
+                        .valid_at(date.today())
+                        .first())
+                elif kind == 'region':
+                    counter['regions'] += 1
+                    # Only link to pre-2016 regions which kept the same id.
+                    new_zones.append(
+                        GeoZone
+                        .objects(code=zone_id, level='fr:region')
+                        .first())
+                elif kind == 'epci':
+                    counter['epci'] += 1
+                    new_zones.append(
+                        GeoZone
+                        .objects(code=zone_id, level='fr:epci')
+                        .latest_valid())
```

Comment:
I agree but I think that if I go this path, I need to do the same for all zones (ie. choose town valid at dataset submission instead of town valid today)